app/assets/javascripts/uploadcare/settings.coffee

```diff
@@ -201,6 +203,16 @@ uploadcare.namespace 'settings', (ns) ->
 
     if settings.validators
       settings.validators = settings.validators.slice()
+    
+    if settings.previewUrlProvider and not settings.overridePreviewUrl
+      settings.overridePreviewUrl = (url, info) => 
+        useGetParam = /\?$/.test(settings.previewUrlProvider)
+        location = if useGetParam then "url=#{encodeURIComponent(url)}" else "/#{url}"
+        utils.normalizeUrl(settings.previewUrlProvider) + location
+    
+    if settings.overridePreviewUrl
+      if typeof settings.overridePreviewUrl is 'string'
+        settings.overridePreviewUrl = eval(settings.overridePreviewUrl)
```

Comment:
no eval, please

Reply:
There is no any other way if we want widget to accept attribute like `data-overrider-preview-url="window.someFunction"`